### ISSUE TITLE
[Bug]: Tips and invoiced amount are inseparable in csv

### ISSUE THREAD

**Issue by Liongrass** [btcpayserver/btcpayserver#5290]
### What is your BTCPay version?

`BTCPay Server v1.11.4+79b2f1652`

### How did you deploy BTCPay Server?

Lunanode deployment

### What happened?

I'm unable to separate tips from total amounts in the csv.

### How did you encounter this bug?

- I, the admin, enable tipping for a store, using the Keypad-only checkout
- The cashier enters the amount of the service, e.g. $1
- The customer enters the amount of the tip, e.g. 20%
- The customer pays the invoice
- I download the csv for the store
**- I'm unable to find out how much the service cost, how high of a tip the customer paid**

This makes it very hard to run a store with tipping enabled. How am I supposed to pay out the tips to the servers at the end of the day if I don't know how many tips were paid?

### Are you sure this is a bug report?

- [X] I confirm this is a bug report

**Comment by NicolasDorier** [comment]
@Liongrass to make sure, you mean in the Reporting/Products sold?

**Comment by NicolasDorier** [comment]
The problem is that the "Products Sold" report is 1 line = 1 product. As such we can't really put tip information in there.

In which report you'd like this information?

**Comment by Liongrass** [comment]
Yeah, good question how this can be surfaced ideally without messing with the document structure, which is already a bit confusing. One idea I had was to use `InvoicePrice` as the amount pre-tip, and `PaidCurrency` as the amount post-tip, then the difference can be the tip. But we'd have to check if that messes up something else.

**Comment by Liongrass** [comment]
My idea: the tip and the payment are recorded in different rows, with the same details but a different order ID. So a $1 payment with a 20% tip shows up twice in the csv, once as the $1 payment, and once as the $0.2 tip.

**Comment by NicolasDorier** [comment]
Can't do this, order id is tied to an invoice, and several payments can be for same invoice.
There is no concept of orderId outside of an invoice.

**Comment by Liongrass** [comment]
Given other issues with the csv [4936](https://github.com/btcpayserver/btcpayserver/issues/4936), maybe this mechanism has to be rethought or overhauled? It currently seems very difficult to use BTCPay for accounting unless fully integrating external software through the API.

**Comment by NicolasDorier** [comment]
I wished I could find a better mechanism, but so far nobody told me how it should look like.
What is the problem of adding the TIP % as an additional column of Payments report?

**Comment by Liongrass** [comment]
Yeah, that seems to be the minimum information required to make this useful. Maybe reporting the tip as an absolute number would make it more straight forward to sum it up using the spreadsheet alone.

**Comment by NicolasDorier** [comment]
How about another idea: Add it "Products sold".

**Comment by Liongrass** [comment]
Yeah as long as that will also work for the keypad-pos. Looking at my store under "products sold" currently shows an empty list. Ideally it would also divide it by App, because I might have a separate app for each attendant/server, and they wouldn't necessarily share tips.

**Comment by Liongrass** [comment]
The scope of the csv-export of the summary of "products sold" would have to be expanded too in that case, as for example the total of sats collected matters too, not just the dollars invoiced.

**Comment by dstrukt** [comment]
cc @dennisreimann relevant feedback for us, and the next design call

**Comment by Fustler** [comment]
I'm just adding my support to have this detail rectified. Here in Vancouver, we are onboarding a healthy amount of businesses. One of the businesses is where I host my meet ups.  They sell drinks and the bartenders need to accept tips.  Currently we end up having to either add x% to the cost of the drinks and then manually do the math and deduct the x% from the total to determine the tip. Or we create a secondary QR code/wallet where the customer can do a second transaction to keep tips separate from the regular sales.  Both aren't a great experience. If the reporting can break out the cost of the product/service and the tip separately, this will go a LONG way in helping onboard more businesses. Thank you!

**Comment by dennisreimann** [comment]
@Fustler  We have that data structured in a form that we can separate them, so this is good feedback and we will change it to be separate columns.

**Comment by Kukks** [comment]
we can experiment using the dynamic reports, which is able to parse the
json metadata  in SQL dirctly

**Comment by pavlenex** [comment]
We plan to improve and in a way redefine reporting, closing this issue since the tracker is here, if anyone has specific feedback on what they need from reporting, feel free to comment there. https://github.com/btcpayserver/btcpayserver/issues/6716
